tck/methods/account.go

```diff
@@ -0,0 +1,279 @@
+}
+type AccountInfoResponse struct {
+	AccountID                     string
+	Balance                       string
+	Key                           string
+	AccountMemo                   string
+	MaxAutomaticTokenAssociations uint32
+	AutoRenewPeriod               time.Duration
+}
+
+// Variable to be set to `SetGrpcDeadline` for all transactions
+var threeSec = time.Second * 3
+
+// SetSdkService We set object, which is holding our client params. Pass it by referance, because TCK is dynamically updating it
+func (a *AccountService) SetSdkService(service *SDKService) {
+	a.sdkService = service
+}
+
+// TODO Find better way to handle error. We want some wrapper, which would intercept the error and contruct the correct
+// TODO jrpc error based on that  "jrpc2.Errorf(-32603, "Internal error"/"Hedera error", ErrorStatus{Status: err.Error()})"
+// CreateAccount jRPC method for createAccount
+func (a *AccountService) CreateAccount(_ context.Context, accountParams CreateAccountParams) (*AccountResponse, error) {
+	transaction := hedera.NewAccountCreateTransaction().SetGrpcDeadline(&threeSec)
+
+	if accountParams.PublicKey != "" {
+		key, err := hedera.PublicKeyFromString(accountParams.PublicKey)
+		if err != nil {
+			return nil, jrpc2.Errorf(-32603, "Internal error", ErrorStatus{Status: err.Error()})
+		}
+		transaction.SetKey(key)
+	}
+	if accountParams.InitialBalance != 0 {
+		transaction.SetInitialBalance(hedera.HbarFromTinybar(accountParams.InitialBalance))
+	}
+	if accountParams.ReceiverSignatureRequired {
+		transaction.SetReceiverSignatureRequired(accountParams.ReceiverSignatureRequired)
+	}
+	if accountParams.MaxAutomaticTokenAssociations != 0 {
+		transaction.SetMaxAutomaticTokenAssociations(accountParams.MaxAutomaticTokenAssociations)
+	}
+	if accountParams.StakedAccountId != "" {
+		accountId, err := hedera.AccountIDFromString(accountParams.StakedAccountId)
+		if err != nil {
+			return nil, jrpc2.Errorf(-32603, "Internal error", ErrorStatus{Status: err.Error()})
+		}
+		transaction.SetStakedAccountID(accountId)
+	}
+	if accountParams.StakedNodeId != 0 {
+		transaction.SetStakedNodeID(accountParams.StakedNodeId)
+	}
+	if accountParams.DeclineStakingReward {
+		transaction.SetDeclineStakingReward(accountParams.DeclineStakingReward)
+	}
+	if accountParams.AccountMemo != "" {
+		transaction.SetAccountMemo(accountParams.AccountMemo)
+	}
+	if accountParams.AutoRenewPeriod != 0 {
+		transaction.SetAutoRenewPeriod(accountParams.AutoRenewPeriod)
+	}
+	if accountParams.PrivateKey != "" {
+		key, err := hedera.PrivateKeyFromString(accountParams.PrivateKey)
+		if err != nil {
+			return nil, jrpc2.Errorf(-32603, "Internal error", ErrorStatus{Status: err.Error()})
+		}
+
+		_, err = transaction.FreezeWith(a.sdkService.Client)
+		if err != nil {
+			return nil, jrpc2.Errorf(-32603, err.Error())
+		}
+		transaction.Sign(key)
+	}
+
+	txResponse, err := transaction.Execute(a.sdkService.Client)
+	if err != nil {
+		fmt.Println(err)
```

Comment:
Guessing this came for some sort of debugging. Should be removed.

Reply:
There are others in this file.

Reply:
Refactored the error handling